### ISSUE TITLE
Regex Match Error when parsing '---'

### ISSUE THREAD

**Issue by kenlimmj** [jxson/front-matter#30]
### Problem
Attempting to extract front matter from a string containing only `---` throws the error "Cannot read property 'length' of null". Encountered this because I was creating a markdown editor that accepts YAML headers, but when I started typing the opening line `---`, kittens died :(

The problem appears to be with the following function:
```javascript
function parse(string) {
  var match = regex.exec(string)
  var yaml = match[match.length - 1].replace(/^\s+|\s+$/g, '')
  var attributes = parser.load(yaml) || {}
  var body = string.replace(match[0], '')

  return { attributes: attributes, body: body }
}
```
`regex.exec` returns `null` if nothing in the requested match fails, but the next line assumes that `match` is defined by calling `match.length`.

### Steps to Reproduce
See [here](https://tonicdev.com/kenlimmj/front-matter-bug) for a MWE.

### Proposed Fix
Guard the regex match with a nullity check defaulting to the empty string:
```javascript
var match = regex.exec(string) || '';
```

**Comment by jxson** [comment]
Thanks for the excellent bug report. I will take a look at adding a fix in the next day or so.

**Comment by jxson** [comment]
@kenlimmj just published a fix with 2.0.3.

When I originally created the module I didn't think about things like live updating, its a neat use case! Let me know if you run into any other issues...
